src/test/java/com/google/cloud/flink/bigquery/integration/FlinkBQWriteIntegrationTest.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.flink.bigquery.integration;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.apache.flink.table.api.Expressions.$;
+
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.flink.bigquery.BigQueryTableSink;
+import java.io.IOException;
+import org.apache.flink.api.common.JobExecutionResult;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CloseableIterator;
+import org.junit.Test;
+
+public class FlinkBQWriteIntegrationTest extends FlinkBigQueryIntegrationTestBase {
+  private static final String PROJECT_ID = ServiceOptions.getDefaultProjectId();
+
+  @Test
+  public void testWriteToBigQuerySupportsOverwrite() throws Exception {
+    String bigqueryReadTable = "bigquery-public-data.samples.shakespeare";
+    String bigqueryWriteTable =
+        PROJECT_ID + "." + testDataset.toString() + "." + "supportsOverwrite";
+    testWriteToBigQuery(bigqueryReadTable, bigqueryWriteTable);
+    Thread.sleep(120 * 1000);
```

Comment:
Why do we have to sleep for such a long time?

Reply:
It takes this amount of time to write the data.